### PR TITLE
Return error to the user if authentication failed

### PR DESCRIPTION
**Pull request by khaledkbadr**
Signed-off-by: Khaled K. Badr <khaledkbadr@outlook.com>

### CODE REVIEW

File: api/tools/connection.go [hunk 1]
```diff
@@ -139,16 +136,16 @@ func ConnectionMiddleware(opt ...ConnectionOptions) func(h http.Handler) http.Ha
...
-func GetAysConnection(r *http.Request, api API) AYStool {
+func GetAysConnection(r *http.Request, api API) (AYStool, error) {
 	aysAPI := api.AysAPIClient()
 
 	token, err := api.GetJWT()
 	if err != nil {
-		log.Error(err.Error())
-	} else {
-		aysAPI.AuthHeader = fmt.Sprintf("Bearer %s", token)
+		return GetAYSClient(aysAPI), err
```
Comment (FastGeert) [inline api/tools/connection.go]: Should return nil, err

File: api/tools/jwt.go [hunk 2]
```diff
@@ -88,8 +87,12 @@ func (jp *JWTProvider) refreshToken() (string, error) {
 }
 
 func (jp *JWTProvider) GetJWT() (string, error) {
+	if jp.development {
+		return jp.jwt, nil
+	}
+
 	now := time.Now().Unix()
-	if rem := jp.expires - now; rem < 300 {
+	if rem := jp.expires - now; rem < 3600 {
```
Comment (FastGeert) [inline api/tools/jwt.go]: ??? JWT are refreshed for 3600 seconds, which always result in less than 3600 seconds remaining. The original code is correct.

File: api/tools/jwt.go [hunk 3]
```diff
@@ -88,8 +87,12 @@ func (jp *JWTProvider) refreshToken() (string, error) {
 }
 
 func (jp *JWTProvider) GetJWT() (string, error) {
+	if jp.development {
+		return jp.jwt, nil
```
Comment (FastGeert) [inline api/tools/jwt.go]: Shouldn't we return "", nil ??? there is no jwt, so

Comment (rkhamis) [reply api/tools/jwt.go]: It's already going to be empty, since we get the DevelmentJWTProvider

File: api/tools/connection.go (hunk 1 above)
Comment (khaledkbadr) [reply api/tools/connection.go]: The method return `AYStool` object, so I thought to return the default one. If you have an idea how to go around that to return nil Im all for it

Comment (FastGeert) [reply api/tools/connection.go]: The nil is for the error case

File: api/tools/jwt.go (hunk 3 above)
Comment (FastGeert) [reply api/tools/jwt.go]: indeed, but it does not make sense to return jp.jwt. People that will read the code later on, will then clearly see that it always returns "" in development mode.

**Pull request merged**